### DISCUSSION TITLE
Value types & Reference types

### DISCUSSION THREAD

**Discussion by gabrielbuenoalves** [balta-io/forum#513] [Tira Dúvidas] [answered]
Uma dúvida em questão a memória Heap e Stack. A memória Heap armazena os dados
enquanto Stack armazena as referências dos dados, logo os Value types são armazenados na memória Heap enquanto os Reference types são armazenados na memória Stack?

**Comment by henriqueweege** [comment, marked as answer]
Oi, Gabriel! Tudo bem?
A resposta rápida é não. Value Types contém  sua propria cópia dos dados e são armazenados na Stack. Reference Types contém apenas referências a dados, e embora os ponteiros das referências vão para a Stack, os dados de fato estão na Heap.
Se eu não tiver sido claro, encontrei [esse artigo ](
https://www.linkedin.com/pulse/value-vs-reference-types-c-roman-fairushyn?utm_source=share&utm_medium=member_android&utm_campaign=share_via) que pode ser que te ajude.

Valeuuuu

**Reply by andrebaltieri** [reply]
👏
